### ISSUE TITLE
Error when trying to use gridfs

### ISSUE THREAD

**Issue by JarnoRFB** [mongomock/mongomock#526]
From what I have read in #311 seems to be somehow there, but some people still encounter errors. I tried to use the gridfs integration like
```python
import gridfs
from mongomock.gridfs import enable_gridfs_integration
enable_gridfs_integration()

client = mongomock.MongoClient()
db = client.db
runs = db.runs
metrics = db.metrics
fs = gridfs.GridFS(db)
```

However, then I get the error

```python
../../../venv/lib/python3.5/site-packages/gridfs/__init__.py:65: in __init__
    database = _clear_entity_type_registry(database)
../../../venv/lib/python3.5/site-packages/gridfs/grid_file.py:103: in _clear_entity_type_registry
    codecopts = entity.codec_options.with_options(type_registry=None)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = Collection(Database(mongomock.MongoClient('localhost', 27721), 'db'), 'codec_options')
kwargs = {'type_registry': None}, forbidden_kwargs = set()

    def with_options(self, **kwargs):
        forbidden_kwargs = set(kwargs.keys()) - set(_WITH_OPTIONS_KWARGS)
        if forbidden_kwargs:
            raise TypeError(
>               "with_options() got an unexpected keyword argument '%s'" % forbidden_kwargs.pop())
E           TypeError: with_options() got an unexpected keyword argument 'type_registry'
```

Is there any way around this? I am using mongomock version 3.16 and Python 3.5. Thanks in advance!

**Comment by gst** [comment]
I get the exact same error but I'm using 
+ python 3.6(.7) 
+ mongomock 3.16.0 
+ pytest-mongodb 2.1.2

**Comment by FerumFlex** [comment]
Tried downgrade pymongo to 3.7.2 and it works

**Comment by pcorpet** [comment]
I think this is fixed by 14a87ee5 so using `v3.17.0` of mongomock should work. Please reopen if I'm wrong.

**Comment by anatoly-kor** [comment]
Hi everyone. I have the same bug. It looks like what bug in differences **pymongo.database.Database** and **mongomock.Database**. That's the reason for `TypeError: database must be an instance of Database` for GridFS.

```python
import mongomock
import pymongo

db = pymongo.MongoClient().db_example

print(isinstance(db, pymongo.database.Database))  # True
print(isinstance(db, mongomock.Database))    # False
```
